### ISSUE TITLE
result returns prediction for classes they never been trained

### ISSUE THREAD

**Issue by Thaispookie** [thtrieu/darkflow#839]
I follow the "Training on your own dataset" instruction with 3 Classes.
As example
Dog
Birth
Cat

After I train only the dogs but the result = tfnet.return_predict(img) returns perdictions for non trained classes.

Was is going wrong here anyone has please an advice.

Thanks Roman

**Comment by jatinmandav** [comment]
Hi @Thaispookie !
Can you show your cfg file? And what do you exactly mean by "only the dogs"? Are you training one class at a time?

**Comment by Thaispookie** [comment]
@jatinmandav 

Yes I mean that train only one class  "dog" and the cfg is as attachment.

Thanks

Roman

[tiny-yolo-voc-3c.zip](https://github.com/thtrieu/darkflow/files/2187709/tiny-yolo-voc-3c.zip)

**Comment by jatinmandav** [comment]
If your cfg files shows 3 classes, may I ask why are you only training it for one class? Try to train with all the 3 classes and see if problem still persists. :)

**Comment by ollfa** [comment]
I have a similar problem. 

If I train with yolov2-tiny-voc.weights on my own dataset flow outputs "Parsing for ['person','vehicle']", which are my two classes.  (I have changed cfg to classes = 2 and filters = 35).

Mostly it returns person/vehicle, but sometimes it classifies cat/dog/tvmonitor or some other label from the original labels.txt.
